File: infrastructureservice/PlatformCI.InfrastructureService.Models/Implementation/InfrastructureDescriptor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using PlatformCI.InfrastructureService.Models.Interfaces;

namespace PlatformCI.InfrastructureService.Models.Implementation
{
    [Serializable]
    public class InfrastructureDescriptor : IRepositorySolutionProjectKey
    {
        public IList<IisSite> IisSites { get; set; }
        public IList<IisApplication> IisApplications { get; set; }
        public IList<IisApplicationPool> IisApplicationPools { get; set; }
        public IList<WindowsService> WindowsServices { get; set; }
        public IList<ScheduledTask> ScheduledTasks { get; set; }
        public IList<string> ApplicableEnvironments { get; set; }
        public string RepositoryName { get; set; }
        public string SolutionName { get; set; }
        public string ProjectName { get; set; }

        public bool AnyPhysicalInfrastructure()
        {
            return IisSites.Any()
                   || IisApplications.Any()
                   || WindowsServices.Any()
                   || ScheduledTasks.Any();
        }
    }
}

File: models/inframodel/infrastructuredescriptor.go
package inframodel

type InfrastructureDescriptor struct {
	IisSites               []IisSite
	IisApplications        []IisApplication
	IisApplicationPools    []IisApplicationPool
	WindowsServices        []WindowsService
	ScheduledTasks         []ScheduledTask
	ApplicableEnvironments []string
	RepositoryName         string
	SolutionName           string
	ProjectName            string
}

func (i InfrastructureDescriptor) AnyPhysicalInfrastructure() bool {
	return len(i.IisSites) > 0 ||
		len(i.IisApplications) > 0 ||
		len(i.WindowsServices) > 0 ||
		len(i.ScheduledTasks) > 0
}
